intro/classes/index.html

```diff
@@ -185,19 +152,165 @@ <h3>Single leading underscores</h3>
+
+{% highlight python %}
+class NPC(Actor):
+    def __init__(self, position, image_path, size, dialogue):
+            Actor.__init__(self, position, image_path, size)
+            self.dialogue = dialogue
+{% endhighlight %}
+
+            Calling the superclass constructor:
+
+{% highlight python %}
+class NPC(Actor):
+    def __init__(self, position, image_path, size, dialogue):
+            super().__init__(position, image_path, size)
+            self.dialogue = dialogue
+{% endhighlight %} 
+        </p>
+        
+        <p>
+            These look practically identical (and in fact, they practically do the exact same thing in simple cases), but there are a
+            few reasons to choose one over the other. Generally, <code>super().__init__</code> is preferred because it's easy to swap
+            out or rename the superclass if you need to, as well as not needing to remember the <code>self</code> parameter. And for
+            longer class names, it's also shorter. But, the situation where you would possibly want to use <code>Actor.__init__</code>
+            would be in the case of more complicated inheritance structures (long chains of inheritance or multiple inheritance,
+            both of which are well outside the scope of this article and game design in general, but a link is at the bottom of the
+            page for an excellent video on the topic from the 
+            <a href="https://www.youtube.com/c/mCodingWithJamesMurphy" class="new-tab">mCoding</a> YouTube channel).
+        </p>
+
+        <p>
+            The power of class inheritance is that attributes are <i>inherited</i> from the superclass to the subclass. So, an instance of
+            <code>NPC</code> as given above has the same attributes as <code>Actor</code> (note that this would not be the case if we had
+            not called the constructor of the superclass). Here's an example to play around with:
+            
+            <iframe width="800px" height="200px" class="python-embed" src="https://godbolt.org/e#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:python,selection:(endColumn:16,endLineNumber:10,positionColumn:16,positionLineNumber:10,selectionStartColumn:16,selectionStartLineNumber:10,startColumn:16,startLineNumber:10),source:'class+Actor:%0A++++def+__init__(self,+name):%0A++++++++self.name+%3D+name%0A%0Aclass+NPC(Actor):%0A++++def+__init__(self,+name):%0A++++++++super().__init__(name)%0A%0Anpc+%3D+NPC(%22Tom%22)%0Aprint(npc.name)'),l:'5',n:'0',o:'Python+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:executor,i:(argsPanelShown:'1',compilationPanelShown:'0',compiler:python310,compilerOutShown:'0',execArgs:'',execStdin:'',fontScale:14,fontUsePx:'0',j:1,lang:python,libs:!(),options:'',source:1,stdinPanelShown:'1',tree:'1',wrap:'1'),l:'5',n:'0',o:'Executor+Python+3.10+(Python,+Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4"></iframe>
```

Comment:
I'd suggest adding some additional attribute for `NPC` because otherwise explicit `__init__` declaration is redundant, could also maybe mention that but that might make it more confusing, so, could just add an extra attribute which would also reinforce the understanding of power of inheritance.